### ISSUE TITLE
Safari Issues

### ISSUE THREAD

**Issue by benjaminstrasser** [iden3/snarkjs#73]
When calling fullProve on Safari and Chrome on IOs

`
TypeError: wA.cpus is not a function. (In 'wA.cpus()', 'wA.cpus' is undefined)
`

**Comment by T-Damer** [comment]
Something similar for me:
```
{"message":"Te.cpus is not a function. (In 'Te.cpus()', 'Te.cpus' is undefined)","line":1,"column":221199,"sourceURL":"https://localhost/scripts/snarkjs.min.js"
```

The code in `snark.min.js` is following:
```JS
var Te = {}
navigator.hardwareConcurrency ? navigator.hardwareConcurrency : Te.cpus().length
```

It happens on `Safari` because, it lacks the [Navigator.hardwareConcurrency](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/hardwareConcurrency#browser_compatibility)

We need simple patch like `Te.cpus?.()` (or `wA.cpus?.()` in your case)

The part of the code that causes this (in `snark.js`):
![CleanShot 2023-11-11 at 18 43 13](https://github.com/iden3/snarkjs/assets/49658988/67912a78-5d6e-4e6e-ba96-31108becfe46)

Basically this code suggests, that we have `os` in the browser (but it's available only in node.js). I would like to do a PR, but it's hard for me to build the library properly

It can be fixed by doing
```JS
concurrency os && os.cpus?.() || 1
```

**Comment by OBrezhniev** [comment]
Fixed long time ago...
